tools/config/seed.config.ts

```diff
@@ -78,9 +74,49 @@ export class SeedConfig {
   ];
 
 
-  DEV_DEPENDENCIES = this.DEV_NPM_DEPENDENCIES.concat(this.APP_ASSETS);
-  PROD_DEPENDENCIES = this.PROD_NPM_DEPENDENCIES.concat(this.APP_ASSETS);
+  get PROD_DEPENDENCIES(): InjectableDependency[] {
+    console.warn('The property "PROD_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    return normalizeDependencies(this.NPM_DEPENDENCIES.filter(filterDependency.bind(null, ENVIRONMENTS.PRODUCTION)))
+      .concat(this.APP_ASSETS.filter(filterDependency.bind(null, ENVIRONMENTS.PRODUCTION)));
+  }
+
+  get DEV_DEPENDENCIES(): InjectableDependency[] {
+    console.warn('The property "DEV_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    return normalizeDependencies(this.NPM_DEPENDENCIES.filter(filterDependency.bind(null, ENVIRONMENTS.DEVELOPMENT)))
+      .concat(this.APP_ASSETS.filter(filterDependency.bind(null, ENVIRONMENTS.DEVELOPMENT)));
+  }
+
+  set DEV_DEPENDENCIES(val: InjectableDependency[]) {
+    console.warn('The property "DEV_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+  }
+
+  set PROD_DEPENDENCIES(val: InjectableDependency[]) {
+    console.warn('The property "PROD_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+  }
+
+  get DEV_NPM_DEPENDENCIES(): InjectableDependency[] {
+    console.warn('The property "DEV_NPM_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    return normalizeDependencies(this.NPM_DEPENDENCIES.filter(filterDependency.bind(null, ENVIRONMENTS.DEVELOPMENT)));
+  }
+  get PROD_NPM_DEPENDENCIES(): InjectableDependency[] {
+    console.warn('The property "PROD_NPM_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    return normalizeDependencies(this.NPM_DEPENDENCIES.filter(filterDependency.bind(null, ENVIRONMENTS.PRODUCTION)));
+  }
+  set DEV_NPM_DEPENDENCIES(value: InjectableDependency[]) {
+    console.warn('The property "DEV_NPM_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    const notDev = this.NPM_DEPENDENCIES.filter(d => !filterDependency(ENVIRONMENTS.DEVELOPMENT, d));
+    this.NPM_DEPENDENCIES = notDev.concat(value);
+  }
+  set PROD_NPM_DEPENDENCIES(value: InjectableDependency[]) {
+    console.warn('The property "PROD_NPM_DEPENDENCIES" is deprecated. Consider using "DEPENDENCIES" instead.');
+    const notProd = this.NPM_DEPENDENCIES.filter(d => !filterDependency(ENVIRONMENTS.PRODUCTION, d));
+    this.NPM_DEPENDENCIES = notProd.concat(value);
+  }
 
+  get DEPENDENCIES(): InjectableDependency[] {
+    return normalizeDependencies(this.NPM_DEPENDENCIES.filter(filterDependency.bind(null, this.ENV)))
```

Comment:
Considering the above comment, we could just `bind` to this, the `ENVIRONMENTS` and `ENV` would be available.
Just for the sake of simplifying a bit